substratevm/src/com.oracle.svm.agent/src/com/oracle/svm/agent/restrict/SerializationAccessVerifier.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2019, 2020, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2020 Alibaba Group Holding Limited. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.agent.restrict;
+
+import com.oracle.svm.configure.config.SerializationConfiguration;
+import com.oracle.svm.configure.trace.AccessAdvisor;
+import com.oracle.svm.jni.nativeapi.JNIEnvironment;
+
+public class SerializationAccessVerifier extends AbstractAccessVerifier {
+    private final SerializationConfiguration configuration;
+
+    public SerializationAccessVerifier(SerializationConfiguration configuration, AccessAdvisor advisor) {
+        super(advisor);
+        this.configuration = configuration;
+    }
+
+    public boolean verifyGenerateSerializationConstructor(JNIEnvironment env, String serializationTargetClass, Object parameterTypes, Object checkedExceptions,
+                    int modifiers, String targetConstructorClass) {
```

Comment:
This check will also need to look at the caller to determine if it should be ignored (`shouldApproveWithoutChecks`). The verifiers are part of the agent's _restriction_ mode (as opposed to its tracing mode), I can take care of this part if you prefer.

Reply:
fixed